libs/core/src/main/java/org/opensearch/core/xcontent/MediaTypeParserRegistry.java

```diff
@@ -41,40 +41,39 @@
  *
  * @opensearch.internal
  */
-public class MediaTypeParser<T extends MediaType> {
-    private final Map<String, T> formatToMediaType;
-    private final Map<String, T> typeWithSubtypeToMediaType;
+public final class MediaTypeParserRegistry {
+    private static Map<String, MediaType> formatToMediaType;
+    private static Map<String, MediaType> typeWithSubtypeToMediaType;
 
-    public MediaTypeParser(T[] acceptedMediaTypes) {
-        this(acceptedMediaTypes, Map.of());
-    }
+    // Default mediaType singleton
+    private static MediaType DEFAULT_MEDIA_TYPE;
 
-    public MediaTypeParser(T[] acceptedMediaTypes, Map<String, T> additionalMediaTypes) {
+    public static void register(MediaType[] acceptedMediaTypes, Map<String, MediaType> additionalMediaTypes) {
         final int size = acceptedMediaTypes.length + additionalMediaTypes.size();
-        Map<String, T> formatMap = new HashMap<>(size);
-        Map<String, T> typeMap = new HashMap<>(size);
-        for (T mediaType : acceptedMediaTypes) {
+        Map<String, MediaType> typeMap = new HashMap<>(size);
+        Map<String, MediaType> formatMap = new HashMap<>(size);
+        for (MediaType mediaType : acceptedMediaTypes) {
             typeMap.put(mediaType.typeWithSubtype(), mediaType);
             formatMap.put(mediaType.format(), mediaType);
         }
-        for (Map.Entry<String, T> entry : additionalMediaTypes.entrySet()) {
+        for (Map.Entry<String, MediaType> entry : additionalMediaTypes.entrySet()) {
             String typeWithSubtype = entry.getKey();
-            T mediaType = entry.getValue();
+            MediaType mediaType = entry.getValue();
 
             typeMap.put(typeWithSubtype.toLowerCase(Locale.ROOT), mediaType);
             formatMap.put(mediaType.format(), mediaType);
         }
 
-        this.formatToMediaType = Map.copyOf(formatMap);
-        this.typeWithSubtypeToMediaType = Map.copyOf(typeMap);
+        formatToMediaType = Map.copyOf(formatMap);
+        typeWithSubtypeToMediaType = Map.copyOf(typeMap);
```

Comment:
Seems a bit dangerous to have this public API that can be called at any time and overwrite the existing mappings, no? Is there a way to keep this a bit more locked down for the time being?

Reply:
updated the `register` method to ensure the map is not overwritten and safety check dual registrations.

Reply:
@nknize sorry for jumping late, the implementation is still susceptible to the races, calling `register` is thread-safe from the modification perspective but leads to lost entries (that is fine if we just document it but fixing that is even better).